DirectXHost.cs

```diff
@@ -241,134 +272,159 @@ private void _dxForm_HelpRequested(object sender, EventArgs eventArgs)
-		}
-
-		public void Draw()
-		{
-			_graphics.ClearRenderTargetView();
-			_graphics.PresentSwapChain();
-		}
-
-		#region IDisposable Support
-
-		protected virtual void Dispose(bool disposing)
-		{
-			if (!IsDisposed)
-			{
-				if (disposing)
-				{
-					// TODO: dispose managed state (managed objects).
-					_dxForm.Dispose();
-				}
-
-				_graphics.Dispose();
-				IsDisposed = true;
-			}
-		}
-
-		// TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
-		~DirectXHost()
-		{
-			// Do not change this code. Put cleanup code in Dispose(bool disposing) above.
-			Dispose(false);
-		}
-
-		// This code added to correctly implement the disposable pattern.
-		public void Dispose()
-		{
-			// Do not change this code. Put cleanup code in Dispose(bool disposing) above.
-			Dispose(true);
-			// TODO: uncomment the following line if the finalizer is overridden above.
-			GC.SuppressFinalize(this);
-		}
-		#endregion
-
-	}
+        }
+        private int ColorToBgr(Color color) => (color.B << 16) | (color.G << 8) | color.R;
+        private MainMenu GetMenu() => new MainMenu(new MenuItem[]
+            {
+                new MenuItem("?", _dxForm_HelpRequested),
+                new MenuItem($"{(Settings.overlayClickable ? '☑' : '☐')} Overlay Clickable", (menuItem,e) => {
+                    Settings.overlayClickable = !Settings.overlayClickable;
+                    ShouldShowOverlayFrame(Settings.overlayClickable);
+                    Settings.Save();
+                    (menuItem as MenuItem).Text = $"{(Settings.overlayClickable ? '☑' : '☐')} Overlay Clickable";
+                }),
+                new MenuItem($"{(Settings.savePositions ? '☑' : '☐')} Save Window Positions", (menuItem,e) => {
+                    Settings.savePositions = !Settings.savePositions;
+                    Settings.Save();
+                    (menuItem as MenuItem).Text = $"{(Settings.savePositions ? '☑' : '☐')} Save Window Positions";
+                }),
+                new MenuItem("Transparency Colour", (menuItem,e) =>
+                {
+                    var dialog = new ColorDialog();
+                    dialog.Color = Settings.transparencyKey;
+                    dialog.SolidColorOnly = true;
+                    dialog.AnyColor = true;
+                    dialog.FullOpen = true;
+                    dialog.CustomColors = new int[] { ColorToBgr(Constants.DefaultTransparencyKey), ColorToBgr(Settings.transparencyKey) };
+                    if(dialog.ShowDialog(_dxForm) == DialogResult.OK)
+                    {
+                        Settings.transparencyKey = dialog.Color;
+                        Settings.Save();
+                        ShouldShowOverlayFrame(true);
+                    }
+                }),
+                new MenuItem($"{(Settings.frameRate > 0?Settings.frameRate.ToString():"Unlimited")} FPS", (menuItem, e) => {

```

Comment:
NIT: The ternary operator should have spacing around it for better legibility and the round brackets are unnecessary
```C#
$"{Settings.frameRate > 0 ? Settings.frameRate.ToString() : "Unlimited"} FPS"
```